spec/services/deposit_complete_auditor_spec.rb

```diff
@@ -3,34 +3,91 @@
 require "rails_helper"
 
 RSpec.describe DepositCompleteAuditor do
-  let(:work_still_accessioning) { create(:work_version_with_work_and_collection, state: :depositing, druid: "druid:bc123df4567").work }
-  let(:work_done_accessioning) { create(:work_version_with_work_and_collection, state: :depositing, druid: "druid:bc123df4568").work }
-  let(:collection_still_accessioning) { create(:collection_version_with_collection, state: :depositing, collection_druid: "druid:bc123df4569").collection }
-  let(:collection_done_accessioning) { create(:collection_version_with_collection, state: :depositing, collection_druid: "druid:bc123df4560").collection }
-
   let(:workflow_client) { instance_double(Dor::Workflow::Client) }
 
   before do
     allow(Repository).to receive(:valid_version?).and_return(true)
-    # These works / collections should be ignored.
-    build(:valid_deposited_work_version)
-    build(:work_version)
-    build(:collection_version_with_collection) # Deposited
-    build(:collection_version)
     allow(Dor::Workflow::Client).to receive(:new).and_return(workflow_client)
-    allow(workflow_client).to receive(:active_lifecycle).with(druid: work_still_accessioning.druid, milestone_name: "submitted", version: work_still_accessioning.head.version.to_s).and_return(Time.now)
-    allow(workflow_client).to receive(:active_lifecycle).with(druid: work_done_accessioning.druid, milestone_name: "submitted", version: work_done_accessioning.head.version.to_s).and_return(nil)
-    allow(workflow_client).to receive(:active_lifecycle).with(druid: collection_still_accessioning.druid, milestone_name: "submitted", version: collection_still_accessioning.head.version.to_s).and_return(Time.now)
```

Comment:
This commit also includes a bit of a test refactoring, replacing one mega-test with a bunch of smaller, more targeted tests. I can roll that part back if folks think what we had before was a better way to test this behavior.